testing/regress/hpcc/regression/cleanup.py

```diff
@@ -0,0 +1,101 @@
+def getRegressLogs(mode, logDirPath, startTime, endTime):
+    regressLogFiles = glob.glob(os.path.join(logDirPath, '*.log'))   
+    for logFilePath in regressLogFiles:
+        logFileNamePattern = r'^(thor|roxie|hthor)\.\d{2}-\d{2}-\d{2}-\d{2}-\d{2}-\d{2}\.log(?!.*exclusion)'
+        match = re.search(logFileNamePattern, os.path.basename(logFilePath))
+        baseFilePath = os.path.basename(logFilePath)
+        if match:
+            pathParts = baseFilePath.split('.') 
+            fileDatetimeStr = pathParts[1]
+            dateFormat = "%y-%m-%d-%H-%M-%S"
+            fileDate = datetime.strptime(fileDatetimeStr, dateFormat)
+            startDate = datetime.strptime(startTime, dateFormat)
+            endDate = datetime.strptime(endTime, dateFormat)
+            if startDate <= fileDate <= endDate:
+                getWorkunitDetails(logFilePath, mode)
+
+# Extracts workunit details from log file based on the cleanup mode
+def getWorkunitDetails(logFilePath, mode):
+    with open(logFilePath, 'r') as file:
+        lineList = file.readlines()
+        if mode == "all" or mode == "workunits":
+            for line in lineList:
+                index = line.find("URL")
+                if index != -1:
+                    url = line[index + 3:-1]
+                    deleteWorkunit(url)
+        elif mode == "passed":
+            for i, line in enumerate(lineList):
+                if re.search(r'\bpass\b', line, re.IGNORECASE):
+                    nextLine = lineList[i + 1]
+                    index = nextLine.find("URL")
+                    if index != -1:
+                        url = nextLine[index + 3:-1]
+                        deleteWorkunit(url)
+
+# Deletes workunits
+def deleteWorkunit(url):
+    deletionUrl = url.replace("?Widget=WUDetailsWidget&Wuid", "WsWorkunits/WUDelete.json?Wuids")
+    startIndex = url.find('Wuid=')
+    if startIndex != -1:
+        wuid = url[startIndex:]
+    try:
+        response = requests.post(deletionUrl)
+        jsonResponse = response.json()
+        if response.status_code == 200:
+            if jsonResponse.get("WUDeleteResponse") == {}:
+                logger.info("Workunit %s deleted successfully.", wuid)
+            elif "ActionResults" in jsonResponse["WUDeleteResponse"]:
+                errorMessage = jsonResponse["WUDeleteResponse"]["ActionResults"]["WUActionResult"][0]["Result"]
+                logger.error("Failed to delete workunit %s.\n   URL:%s\n   %s Response status code: %d", wuid, url, errorMessage, response.status_code)
+        else:
+            logger.error("Failed to delete workunit %s.\n   URL:%s\n   Response status code: %d", wuid, url, response.status_code)
+    except requests.exceptions.RequestException as e:
+        logger.error("Error occurred while deleting workunit %s: %s.\n   URL: %s", wuid, str(e), url)
+
+# Main function to instantiate custom logger and initiate workunit detail extraction
+def checkCleanupMode(cleanupMode, startTime, endTime):
```

Comment:
I would consider to rename this function (like 'doCleanup()' )  based on it doesn't do any checks now.

Reply:
Sure, will do